### PR TITLE
Improve Release

### PR DESCRIPTION
**Pull request by bjoernricks**
## What

Refactor release code

## Why

Easier to maintain prepare and release code and explicit handling of to be committed files

## References

DEVOPS-551
DOS-53

## Checklist

- [x] Tests

### CODE REVIEW

File: pontos/release/prepare.py [hunk 1]
```diff
@@ -16,162 +16,274 @@
...
-        release_version: str = get_next_patch_version(terminal)
-    else:
-        release_version: str = args.release_version
+class ReleaseType(Enum):
+    PATCH = "patch"
+    CALENDAR = "calendar"
+    VERSION = "version"
+    # for the future we need
+    # MAJOR = "major"
+    # MINOR = "minor"
```
Comment (y0urself) [inline pontos/release/prepare.py:59]:
```suggestion
    # MINOR = "minor"
    # PRE_RELEASE = "pre-release"
```

File: pontos/version/helper.py [hunk 2]
```diff
@@ -118,3 +73,81 @@ def versions_equal(new_version: str, old_version: str) -> bool:
...
+            release_version = Version(
+                f"{str(today.year  % 100)}.{str(today.month)}."
+                f"{str(current_version.micro + 1)}"
+            )
+        else:
+            release_version = Version(
+                f"{str(today.year  % 100)}.{str(today.month)}."
+                f"{str(current_version.micro)}"
+            )
+        return str(release_version)
```
Comment (y0urself) [inline pontos/version/helper.py:109]:
```suggestion
    today = datetime.today()
    cur_year_abrv = today.year % 100

    if (
        current_version.major < cur_year_abrv
        or current_version.minor < today.month
    ):
        release_version = Version(
            f"{str(cur_year_abrv)}.{str(today.month)}.0"
        )
        return str(release_version)

    if (
        current_version.major == cur_year_abrv
        and current_version.minor == today.month
    ):
        if current_version.dev is None:
            release_version = Version(
                f"{str(cur_year_abrv)}.{str(today.month)}."
                f"{str(current_version.micro + 1)}"
            )
        else:
            release_version = Version(
                f"{str(cur_year_abrv)}.{str(today.month)}."
                f"{str(current_version.micro)}"
            )
        return str(release_version)
```

It is not required to explicitly cast `str(int)` within an `f-string` right? We could remove that for readability.

File: pontos/version/helper.py [hunk 3]
```diff
@@ -118,3 +73,81 @@ def versions_equal(new_version: str, old_version: str) -> bool:
...
+        release_version = Version(
+            f"{str(current_version.major)}."
+            f"{str(current_version.minor)}."
+            f"{str(current_version.micro + 1)}"
+        )
+
+    return str(release_version)
+
+
+def get_next_bugfix_version(release_version: str) -> str:
```
Comment (y0urself) [inline pontos/version/helper.py:138]: Bugfix and patch version would be the same currently?
But I am currently not sure where these functions are used ...

Comment (bjoernricks) [reply pontos/version/helper.py]: Yes I already recognized that too. This is one of the points for *additional refactoring* I've mentioned but didn't wanted to touch yet. IMHO `get_next_bugfix_version` should be dropped in favor of an `get_next_patch_version` function with the additional exception handling.

File: pontos/version/commands.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        if command.project_found():
+            return command.get_current_version()
+
+    raise VersionError("No project settings file found")
+
+
+def update_version(
+    new_version: str, *, develop: Optional[bool] = False
+) -> VersionUpdate:
+    for cmd in COMMANDS:
```
Comment (y0urself) [inline pontos/version/commands.py:54]:
```suggestion
def update_version(
    new_version: str, *, develop: Optional[bool] = False
) -> VersionUpdate:
    """
    Update the Version for the current project in the relevant files
    
    Args:
        new_version (str): The version that will be set for the project
        develop (bool): Add a dev version section to the version string

    Raises:
        VersionError if no project settings (file) can be found.
    """
    for cmd in COMMANDS:
```

Somehow this is not recognized as a string oO

File: pontos/version/commands.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    PythonVersionCommand,
+)
+
+
+def get_current_version() -> str:
+    """
+    Get the current Version for the current project
+
+    Raises:
+        VersionError if not current version could be found.
```
Comment (y0urself) [inline pontos/version/commands.py:40]:
```suggestion
        VersionError if current version can not be found.
```

File: pontos/version/commands.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    raise VersionError("No project settings file found")
+
+
+def update_version(
+    new_version: str, *, develop: Optional[bool] = False
+) -> VersionUpdate:
+    for cmd in COMMANDS:
+        command = cmd()
+        project_file = command.project_file_found()
+        if project_file:
```
Comment (y0urself) [inline pontos/version/commands.py:57]:
```suggestion
        if command.project_found():
```

Should also work, like for `get_current_version()`

**Review by y0urself — APPROVED** [review]

**Pull request merged**